declarative-schedule-doc.md

```diff
@@ -339,6 +340,90 @@ scenarios:
+
+Please be noted that conditional schedule does not work with the new scheduler that uses DEFAULT and FLOWS.
+
+YAML_SCHEDULE_DEFAULT should point to a "default" minimal schedule that has sections.  You can find them in schedule/yast/sle/flows/default.yaml. Sections can have "normal yaml schedule lines" or are just "empty sockets" (indicated by "[]").
+
+```yaml
+# Default ordered sequence of steps to be optionally overwritten for this product
+bootloader:
+  - installation/bootloader_start
+setup_libyui:
+  - installation/setup_libyui
+access_beta: []
+product_selection:
+  - installation/product_selection/install_SLES
+license_agreement:
+  - installation/licensing/accept_license
+
+```
+
+YAML_SCHEDULE_FLOW is a comma separated list of "flow" modules. like schedule/yast/sle/flows
+
+Example: desktop.yaml:
+```yaml
+---
+# Register Desktop Applications Module in Extension and Module Selection
+# Select SLES with GNOME in System Role
+extension_module_selection:
+  - installation/module_registration/register_module_desktop
+system_role:
+  - installation/system_role/accept_selected_role_SLES_with_GNOME
+```
+So, in default.yaml we have
+
+```yaml
+extension_module_selection:
+  - installation/module_registration/skip_module_registration
+```
+And when applying the "flow" desktop.yaml, this will be replaced by:
+
+```yaml
+extension_module_selection:
+  - installation/module_registration/register_module_desktop
+```
+General rule is: 
+Every list of instructions under a key (e.g extension_module_selection) in the flow will overwrite the corresponding list in "default".
+When all flows are applied, we finally apply "YAML_SCHEDULE" which then defines the individual instructions for the test suite.
+
+Example: guided_ext4/ext4.yaml:
+```yaml
+name:           guided_ext4
+description:    >
+  Guided Partitioning installation with ext4 filesystem.
+vars:
+  FILESYSTEM: ext4
+  YUI_REST_API: 1
+schedule:
+  guided_filesystem:
+    - installation/partitioning/guided_setup/select_filesystem_option_ext4
+  default_systemd_target:
+    - installation/installation_settings/validate_default_target
+  system_validation:
+    - console/validate_partition_table_via_blkid
+    - console/validate_blockdevices
+    - console/validate_free_space
+test_data:
+  guided_partitioning:
+    filesystem_options:
+      root_filesystem_type: ext4
+  <<: !include test_data/yast/ext4/ext4.yaml
+```
+
+When combining the three layers
+YAML_SCHEDULE_DEFAULT (machine level, the default)
+YAML_SCHEDULE_FLOWS   (overwrite the above)
+YAML_SCHEDULE         (overwrite the above)
+We end up with a final yaml schedule that should not look different from what we're used so far.
+YAML_SCHEDULE_DEFAULT is usually defined in the job group on machine level (defaults), YAML_SCHEDULE_FLOW and YAML_SCHEDULE are defined in the job group on test suite level.
```

Comment:
same here, no machine level.
we might need to define them in each test suite all three, it depends, because it doesn't apply for the installed system, so all three are defined in the same place in job group configuration.